### PR TITLE
Remove unnecessary "missing snippet" warnings

### PR DESCRIPTION
**Pull request by AlexJerabek**
This PR removes unnecessary snippet warnings from the GenerateDocs build process. It does this by:
- Ignoring warnings from lower versioned requirement sets
- Splitting the office and outlook snippets

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[b653217](https://github.com/OfficeDev/office-js-docs-reference/commits/b653217f0cb7bc2c77645919db071b2d4e6d38ab)_: 

:clock10: **Pending**: waiting for processors ([9](https://ops.microsoft.com/#/statistic) builds ahead of you) 

 :warning: **Docs Build is busy**, currently there are 9 builds ahead of this one, for more information you can view the Build queue graph on the [Docs Portal](https://ops.microsoft.com/#/statistic).

**Comment by AlexJerabek** [comment]
As a proof of usefulness, this PR found issue #560

File: generate-docs/json/outlook/snippets.yaml [hunk 1]
```diff
@@ -6231,14 +2635,6 @@
         "seriesTime": {seriesTimeObject},
         "recurrenceTimeZone": {"name": "Pacific Standard Time", "offset": -480}
     }
-'Office.RoamingSettings:interface':
```
Comment (ElizabethSamuel-MSFT) [inline generate-docs/json/outlook/snippets.yaml]: This should be kept here - it's for an Outlook object

File: generate-docs/json/office/snippets.yaml [hunk 2]
```diff
@@ -4379,1971 +3349,259 @@
...
+        // Set the specified fields for the selected task.
+        function setTaskFields(taskGuid) {
+            var targetFields = [Office.ProjectTaskFields.Active, Office.ProjectTaskFields.Notes];
+            var fieldValues = [true, 'Notes for the task.'];
 
+            // Set the field value. If the call is successful, set the next field.
+            for (var i = 0; i < targetFields.length; i++) {
+                Office.context.document.setTaskFieldAsync(
+                    taskGuid,
+                    targetF
```
Comment (ElizabethSamuel-MSFT) [inline generate-docs/json/office/snippets.yaml]: This belong under Outlook, not Office.

Comment (ElizabethSamuel-MSFT) [reply generate-docs/json/office/snippets.yaml]: This being the Office.RoamingSettings:interface snippet...

**Comment by o365devx** [comment]
Docs Build status updates of commit _[f030ec4](https://github.com/OfficeDev/office-js-docs-reference/commits/f030ec4131c581fe8ec37a873993a992f28e533d)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-19b4efbe)

<a id="19b4efbe"></a>
###
 - **[Error]** `````Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/office-js-docs-reference, working directory: W:\ymuu-s, target branch: master, target branch commit: 42dac893fa0edfec34ac50c46904422200ab9ca1, source commit: f030ec4131c581fe8ec37a873993a992f28e533d.`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C11%5C20%5C0e1871f2-88cb-c16b-6d81-dc80e4f66cef%5CPullRequest%5C201911202141019437-561%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=o2%2FgFjS1y%2FsvtnwmBrzRD00HWDh4ty2LePQk%2FEOsKj0%3D&st=2019-11-20T21%3A36%3A41Z&se=2019-12-21T21%3A41%3A41Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[6f7138d](https://github.com/OfficeDev/office-js-docs-reference/commits/6f7138d0d4fdcb0d39a904504f34713f7292ae4a)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-573bcb80)

<a id="573bcb80"></a>
###
 - **[Error]** `````Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/OfficeDev/office-js-docs-reference, working directory: W:\mxhk-s, target branch: master, target branch commit: 42dac893fa0edfec34ac50c46904422200ab9ca1, source commit: 6f7138d0d4fdcb0d39a904504f34713f7292ae4a.`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C11%5C20%5C0e1871f2-88cb-c16b-6d81-dc80e4f66cef%5CPullRequest%5C201911202157112609-561%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=E9YMePV7IJ0goEfHNJmO6UfXVsPbCufr6Fn9LeZcDgc%3D&st=2019-11-20T21%3A52%3A53Z&se=2019-12-21T21%3A57%3A53Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by AlexJerabek** [comment]
Thanks @ElizabethSamuel-MSFT. I'll fix that RoamingSettings snippet, merge your changes, then merge this PR.

File: docs/code-snippets/outlook-snippets.yaml [hunk 3]
```diff
@@ -2285,7 +2285,7 @@ Office.RecurrenceTimeZone:interface:
         "seriesTime": {seriesTimeObject},
         "recurrenceTimeZone": {"name": "Pacific Standard Time", "offset": -480}
     }
-Office.RoamingSettings:interface:
+Office.RoamingSettings#set:member(1):
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/outlook-snippets.yaml]: @AlexJerabek Just curious why you moved this to the set member?

**Comment by o365devx** [comment]
Docs Build status updates of commit _[56dd5b1](https://github.com/OfficeDev/office-js-docs-reference/commits/56dd5b19198212d8cfb8ed821d31015f331b11e2)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/code-snippets/office-snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/code-snippets/office-snippets.yaml) | :white_check_mark:Succeeded | |
[docs/code-snippets/outlook-snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/code-snippets/outlook-snippets.yaml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/excel_online/office-runtime.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/excel_online/office-runtime.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.apiinformation.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.apiinformation.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.auth.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.auth.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.authoptions.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.authoptions.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.dialog.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.dialog.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.displaywebdialogoptions.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.displaywebdialogoptions.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.storage.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/excel_online/office-runtime/officeruntime.storage.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/excel_online/toc.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/excel_online/toc.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/office/office/office.file.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/office/office/office.file.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/outlook/outlook/office.roamingsettings.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/outlook/outlook/office.roamingsettings.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/outlook_1_1/outlook/office.roamingsettings.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/outlook_1_1/outlook/office.roamingsettings.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/outlook_1_2/outlook/office.roamingsettings.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/outlook_1_2/outlook/office.roamingsettings.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/outlook_1_3/outlook/office.roamingsettings.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/outlook_1_3/outlook/office.roamingsettings.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/outlook_1_4/outlook/office.roamingsettings.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/outlook_1_4/outlook/office.roamingsettings.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/outlook_1_5/outlook/office.roamingsettings.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/outlook_1_5/outlook/office.roamingsettings.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/outlook_1_6/outlook/office.roamingsettings.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/outlook_1_6/outlook/office.roamingsettings.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/outlook_1_7/outlook/office.roamingsettings.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/outlook_1_7/outlook/office.roamingsettings.yml) | :white_check_mark:Succeeded | |
[docs/docs-ref-autogen/outlook_1_8/outlook/office.roamingsettings.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/docs/docs-ref-autogen/outlook_1_8/outlook/office.roamingsettings.yml) | :white_check_mark:Succeeded | |
[generate-docs/GenerateDocs.cmd](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/GenerateDocs.cmd) | :white_check_mark:Succeeded | |
[generate-docs/json/excel_online/office-runtime.api.json](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/excel_online/office-runtime.api.json) | :white_check_mark:Succeeded | |
[generate-docs/json/excel_online}/custom-functions-runtime.api.json](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/excel_online%7D/custom-functions-runtime.api.json) | :white_check_mark:Succeeded | |
[generate-docs/json/office/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/office/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/json/outlook/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/outlook/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/json/outlook_1_1/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/outlook_1_1/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/json/outlook_1_2/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/outlook_1_2/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/json/outlook_1_3/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/outlook_1_3/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/json/outlook_1_4/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/outlook_1_4/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/json/outlook_1_5/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/outlook_1_5/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/json/outlook_1_6/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/outlook_1_6/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/json/outlook_1_7/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/outlook_1_7/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/json/outlook_1_8/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/outlook_1_8/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/json/snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/json/snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/script-inputs/local-repo-snippets.yaml](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/script-inputs/local-repo-snippets.yaml) | :white_check_mark:Succeeded | |
[generate-docs/scripts/midprocessor.ts](https://github.com/OfficeDev/office-js-docs-reference/blob/AlexJ-Snips/generate-docs/scripts/midprocessor.ts) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C11%5C20%5C0e1871f2-88cb-c16b-6d81-dc80e4f66cef%5CPullRequest%5C201911202205461114-561%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=Y90SIx%2F3DfDomNBZol3hefpKLappphNKpCpgVRMwW3c%3D&st=2019-11-20T22%3A27%3A09Z&se=2019-12-21T22%3A32%3A09Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**